### ISSUE TITLE
Analytics Modeling OMAS UTs not running in maven build

### ISSUE THREAD

**Issue by mandy-chessell** [odpi/egeria#5225]
The unit tests for Analytics Modeling OMAS do not seem to be running in the maven build - however they are running in the Gradle build.

This was brought to light by PR #5218 where the unit tests are failing in the Gradle build and not the Maven build.  Manually running the tests shows the same test failures as Gradle sees

**Comment by planetf1** [comment]
Example of failure under gradle build: https://github.com/odpi/egeria/runs/2669034989?check_suite_focus=true (this is now fixed as part of the PR work). 
The failure was in :open-metadata-implementation:access-services:analytics-modeling:analytics-modeling-server:test

The tests are in src/test/java & named *Test.java (a standard pattern the test runner should fine) and are using testng

**Comment by planetf1** [comment]
* In analytics modeling the 'api' module IS running testng tests ok `[INFO] Tests run: 43, Failures: 0, Errors: 0, Skipped: 0` whilst 'server' is not `[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0`
* File patterns & dependencies are the same
Looking at what may be different

**Comment by planetf1** [comment]
On the working test I see:
`[DEBUG] Adding to surefire test classpath: /Users/jonesn/.m2/repository/org/apache/maven/surefire/surefire-testng/3.0.0-M5/surefire-testng-3.0.0-M5.jar
 Scope: null` whilst on the non-running test `[DEBUG] Adding to surefire test classpath: /Users/jonesn/.m2/repository/org/apache/maven/surefire/surefire-junit-platform/3.0.0-M5/surefire-junit-platform-3.
0.0-M5.jar Scope: null`

Jupiter is being pulled in via
```
+- org.mockito:mockito-junit-jupiter:jar:3.9.0:test
[INFO] |  \- org.junit.jupiter:junit-jupiter-api:jar:5.7.1:test
[INFO] |     +- org.apiguardian:apiguardian-api:jar:1.1.0:test
[INFO] |     +- org.opentest4j:opentest4j:jar:1.2.0:test
[INFO] |     \- org.junit.platform:junit-platform-commons:jar:1.7.1:test
```

So the fault was this in the pom.xml, which after commenting out allows the tests to work

```
         <!--dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-junit-jupiter</artifactId>
             <scope>test</scope>
         </dependency-->
```

surefire has some built in rules to determine what to use.... but there's little docs on this, 

It may be tricky to definitively find all places this could occur. Most modules use JUnit5, where this is required,and so this log entry and dependency is correct if mockito in use

Sticking to JUnit5 is probably better for integration in general

We do have many components using testng. 

I've looked through every module reporting no tests to run, where the plugin is active (and not explicitly disabled, as has happened in a few cases where at any point a test is broken). In most cases this is because no tests have been written. I noticed the following where there are tests, but they are not running:

12:45:29,684 [INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ configuration-encrypted-file-store-connector ---
12:45:29,687 [INFO] No tests to run.

12:51:37,788 [INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ glossary-author-view-server ---
12:51:37,790 [INFO] No tests to run.
